esm_runscripts/compute.py

```diff
@@ -19,21 +19,48 @@
 
 class compute(jobclass):
     def __init__(self, config):
-
-        self.relevant_files = ["bin", "config", "forcing", "input", "restart_in"]
-        self.all_files_to_copy = self.assemble_file_lists(config, self.relevant_files)
         # Check for a user defined compute recipe in the setup section of the
         # general section. If nothing is found, recipe_steps should evaluate to
         # None and the default is used
         try:
             setup_name = config["general"]["setup_name"]
             recipe_steps = config.get(setup_name, {}).get("compute_recipe") or config["general"].get("compute_recipe")
         except KeyError:
-            print("Your configuration is incorrect, and should include headings for %s as well as general!" % setup_name)
+            print("Your configuration is incorrect!")
+            print(f"It should include headings for {setup_name} as well as general!")
             sys.exit(1)
         super(compute, self).__init__("compute", recipe_steps=recipe_steps)
         config["general"]["jobclass"] = self
 
+    @staticmethod
+    def assemble_filelists(config):
+        """
+        Assembles the list of files needed for the experiment that shall be
+        copied to the experiment tree. **DOES NOT PERFORM ANY COPYING YET!**.
+        Actual filesystem copying is done by copy_files_to_thisrun (from entire
+        filesystem to experiment tree) and copy_files_to_work (from experiment
+        tree to work folder)
+
+        Parameters
+        ----------
+        config : dict
+            The entire experiment configuration
+
+        Returns
+        -------
+        config : dict
+            The entire experiment configuration
+
+        """
+        self = config["general"]["jobclass"]
+        relevant_files = ["bin", "config", "forcing", "input", "restart_in"]
```

Comment:
Check if any other methods refer to `self.relevant_files`. As it is now, this is just a local variable